App.js

```diff
@@ -1,26 +1,115 @@
-import React from 'react'
-import styled from 'styled-components/native'
+import React, { useState, useEffect } from "react";
+import styled from "styled-components/native";
+import {
+  StyleSheet,
+  Button,
+  Text,
+} from "react-native";
+import { BarCodeScanner } from "expo-barcode-scanner";
 
 const Container = styled.View`
   flex: 1;
   background-color: papayawhip;
   justify-content: center;
   align-items: center;
-`
+`;
 
 const Title = styled.Text`
   font-size: 24px;
   color: palevioletred;
-`
+`;
 
 const App = () => {
+  //state for camera permission
+  const [hasPermission, setHasPermission] = useState(null);
+  //state for determining if camera has scanned barcode
+  const [scanned, setScanned] = useState(false);
+  //storing the barcode information
+  const [data, setData] = useState(0);
+  //storing the fetch data
+  const [food, setFood] = useState({});
+  //section in app
+  const [section, setSection] = useState(0);
+  //storing number of scans made. Workaround for getting the "rescan" to work. When trying this 
+  //with just setting scanned to false after each fetch I got infinite loop
+  const [scans, setScans] = useState(0)
+
+  //function that comes with the barcodescanner to get the data from the scanner
+  useEffect(() => {
+    (async () => {
```

Comment:
Great job with `async await` logic :)
For further explanation, you can check out https://medium.com/javascript-in-plain-english/async-await-javascript-5038668ec6eb